### PR TITLE
entities: improves entities management

### PR DESCRIPTION
**Pull request by zannkukai**
* Updates the `mef-typeahead` component to allow more flexibility and
  deals with `bf:Topic`
* Updates `Contribution` component to display contribution entities.
* Updates subjects display in the `DocumentDetailView` component.
* Renames and updates the `contribution-entity` pipe to `entity-label`
  pipe.

Co-authored-by: Renaud Michotte <renaud.michotte@gmail.com>

### CODE REVIEW

File: projects/admin/src/app/classes/typeahead/mef-typeahead.ts [hunk 1]
```diff
@@ -54,108 +57,103 @@ export class MefTypeahead implements ITypeahead {
...
+    return this._http.get(this._apiProxyEntryPoint, {params, headers}).pipe(
+      map((data: any) => {
+        const label = data?.metadata?.authorized_access_point;
+        return `
+          <a href="${value}" target="_blank">${label} <i class="fa fa-external-link"></i></a>
+          <small class="badge badge-secondary ml-1">${source.toUpperCase()}</small>
+        `
+      }),
+      catchError(e => {
+        console.error(`getValueAsHTML :: Unable to resolve ${value}`, e);
```
Comment (Garfield-fr) [inline projects/admin/src/app/classes/typeahead/mef-typeahead.ts:75]: Is this necessary?

File: projects/shared/src/lib/pipe/entity-label.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  private _defaultPartSeparator = ' - '
+
+  /**
+   * Constructor
+   * @param _translateService - TranslateService
+   */
+  constructor(
+    private _translateService: TranslateService
+  ) {
+  }
```
Comment (Garfield-fr) [inline projects/shared/src/lib/pipe/entity-label.pipe.ts:37]: COSMETIC: `constructor(private _translateService: TranslateService) { }` (one line)

File: projects/shared/src/lib/pipe/entity-label.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  /**
+   * Pipe transformation method.
+   *   Get the best possible entity label depending on the current locale language used.
+   *   If entity has some `subdivisions` parts, they will be join using after the main label.
```
Comment (Garfield-fr) [inline projects/shared/src/lib/pipe/entity-label.pipe.ts:42]: Extra space

File: projects/shared/src/lib/view/contribution/contribution.component.ts [hunk 4]
```diff
@@ -15,23 +15,49 @@
...
+  private _entityTypes = this._appSettingsService.settings.agentAgentTypes || {};
+
+  // CONSTRUCTORS & HOOKS =====================================================
+  /**
+   * Constructor
+   * @param _appSettingsService - AppSettingsService
+   */
+  constructor(
+    private _appSettingsService: AppSettingsService
+  ) { }
```
Comment (Garfield-fr) [inline projects/shared/src/lib/view/contribution/contribution.component.ts:49]: COSMETIC: `constructor(private _appSettingsService: AppSettingsService) { }` (one line)

**Review by Garfield-fr — APPROVED** [review]

File: projects/shared/src/lib/pipe/entity-label.pipe.ts (hunk 3 above)
Comment (zannkukai) [reply projects/shared/src/lib/pipe/entity-label.pipe.ts:42]: it's a conscious choice. I think the docstring is more readable that way : A main docstring line, additional lines give more explainations.

**Review by lauren-d — APPROVED** [review]

**Pull request merged**